modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/CppPistacheServerCodegen.java

```diff
@@ -334,17 +337,19 @@ public String apiFilename(String templateName, String tag) {
         String result = super.apiFilename(templateName, tag);
 
         if (templateName.endsWith("impl-header.mustache")) {
-            int ix = result.lastIndexOf(File.separatorChar);
-            result = result.substring(0, ix) + result.substring(ix, result.length() - 2) + "Impl.h";
-            result = result.replace(apiFileFolder(), implFileFolder());
+            result = injectImplInFilename(result, ".h");
         } else if (templateName.endsWith("impl-source.mustache")) {
-            int ix = result.lastIndexOf(File.separatorChar);
-            result = result.substring(0, ix) + result.substring(ix, result.length() - 4) + "Impl.cpp";
-            result = result.replace(apiFileFolder(), implFileFolder());
+            result = injectImplInFilename(result, ".cpp");
         }
         return result;
     }
 
+    private String injectImplInFilename(String filename, String suffix) {
```

Comment:
What about adding a 1-liner docstring (ideally with an example) explaining what this function does?

Reply:
Yep, I'll add the suggested docstring.

Reply:
Thinking about a comment I realized that he method could be named differently. Renamed the method to `implFilenameFromApiFilename`  in a follow-up commit 37c56049ccaa0a37ed2e9c421f4e5b27249ee439 and commented it. 
@wing328  Thank you, for the very valuable feedback!